### DISCUSSION TITLE
Need help with image upload to pocketbase.

### DISCUSSION THREAD

**Discussion by devlop009** [pocketbase/pocketbase#5842] [Q&A]
I've a form in Appsmith and when click submit I want to add the images and 2 text feelds to a collection named submission. The problem I am having is the images are not bing uploaded. 

Image column is capable of accepting multiple images. I tried posting it with both base64 and binary data formates. No filters for what type collection can accept. I am getting "Failed to load the submitted data due to invalid formatting.", "image: The field contains unknown filenames.." 

I am sending a POST request I've tried both json and multipart form data. 

This is the full error I am getting 

`{
  "id": "yttny9tm5yrw106",
  "created": "2024-11-12 08:44:21.892Z",
  "updated": "2024-11-12 08:44:21.892Z",
  "data": {
    "auth": "guest",
    "details": "image: The field contains unknown filenames..",
    "error": "Failed to load the submitted data due to invalid formatting.",
    "execTime": 74.290743,
    "method": "POST",
    "referer": "",
    "remoteIp": "172.17.0.1",
    "status": 400,
    "type": "request",
    "url": "/api/collections/property/records",
    "userAgent": "ReactorNetty/1.1.22",
    "userIp": "x.x.x.x"
  },
  "message": "POST /api/collections/property/records",
  "level": 8
}`

Can someone please tell me what I am doing wrong here? 

Thank you very much in advance.

**Comment by ganigeorgiev** [comment]
There are not enough details to understand how actually you are constructing and submitting the request. You've mentioned that you've tried submitting _"binary data formates"_ but there are no examples showing how.

In general, try to open the browser's "Inspect tool > Network" and copy the sent request body.

I personally don't have experience with Appsmith and I don't think I can help you with it but note that submitting files as base64 is not supported and file upload in PocketBase is possible only via `multipart/form-data` requests.

**Reply by devlop009** [reply]
Sorry for rambeling been working on it for like past 18 hours. 

Just a followup question data formate for the images don't matter? Base64 and binary both sould work correct?

**Reply by ganigeorgiev** [reply]
No, base64 is NOT supported.

If you are manually constructing a JavaScript [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) body then the file(s) must be either a [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) or [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File).

**Reply by devlop009** [reply]
That's the direction I was leaning towards Thank you for the help!
